File: ruby/AlphabeticalAddition.rb
def add_letters(*letters)
    numbers = (1..25).to_a << 0
    range_letter = ('a'..'z').to_a
    hash = Hash[range_letter.zip numbers]
    total = letters.map { |letter| hash[letter] }.inject { |total, num|  total + num}
    if letters == []
        return 'z'
    else 
        if total > 26 
            while total >= 26 
                total -= 26
            end
            return hash.key(total)
        else
            return hash.key(total)
        end
        
    end

    # hash = Hash[range_letter.zip numbers]

# [1,10,19,6,0,16].inject { |t,n| t + n} - 26 - 26
end


p add_letters('a', 'b', 'c')
p add_letters("y", "c", "b") # => 'd' 4
p add_letters("a", "j", "s", "f", "z", "p") # => 'z' 0 


# "a", "j", "s", "f", "z", "p"

[1,10,19,6,0,16].inject { |t,n| t + n} - 26

File: javaScript/Alphabetical Addition.js
function addLetters(...letters) {
    let arrLetters = range('a', 'z')
    let arrNums = [...Array(26).keys()]
    let hash = {}
    let total = 0
    arrNums.push(arrNums.shift())
    arrLetters.forEach((key, i) => { hash[key] = arrNums[i] })
    let equivalente = letters.map((letter) => {
        return hash[letter]
    })
    
    equivalente.forEach((num) => {
        total += num
    })

    while(total > 25){
        total -= 26
    }

    return getKeyByValue(hash, total)

}

function range(start, stop) {
    let result = []
    for(let idx = start.charCodeAt(0), end=stop.charCodeAt(0); idx <=end; ++idx){
        result.push(String.fromCharCode(idx))
    }
    return result
}

function getKeyByValue(obj, value) {
    return Object.keys(obj).find((key) => obj[key] === value)
}

console.log(addLetters('a', 'b', 'c')) // 'f'
console.log(addLetters('a', 'b')) // 'c'
console.log(addLetters('z'))  // 'z'
console.log(addLetters('z', 'a'))  // 'a'
console.log(addLetters('y', 'c', 'b')) // 'd'
console.log(addLetters()) // 'z'



